spec/message_format_spec.rb

```diff
@@ -26,6 +30,50 @@
       expect(message).to eql('Simple string with nothing special')
     end
 
+    it 'formats tags' do
+      pattern = 'Simple string with <A>tags</A>'
+      message = MessageFormat.new(pattern, 'en-US').format({ A: lambda { |content| "<a href=\"https://google.com\">#{content}</a>" } })
```

Comment:
Do we have anything in place to prevent user submitted content from doing XSS? For example, if content is:
~~~
My Template</a><script>alert('i am evil');</script>
~~~

Reply:
I could add something. The content should be controlled by us, for example in this case content will just be `tags`. But if we have variables inside maybe it would be good to check

Reply:
What would be the best way to go about this? Is there a built in sanitize or something?

Reply:
I believe Rails uses https://api.rubyonrails.org/classes/ERB/Util.html#method-c-html_escape

Reply:
If you don't want to depend on Rails, https://ruby-doc.org/stdlib-3.1.0/libdoc/cgi/rdoc/CGI/Escape.html might be better

Reply:
CGI works the best I think for this case. Added it so that any interpreted variables get escaped.

Reply:
I just realized this is probably better suited being done in elaine, kind of like we are already doing in the translation helper. What do you think about this gem not worrying about that and let elaine handle the html escaping

Reply:
Its fine with me. My suggestion is to give it a try in Elaine and see what it looks like. If this thing can return HTML, then someone is going to have to call [html_safe](https://apidock.com/rails/String/html_safe) on the string or else Rails will escape the whole thing and none of the markup will work. Whoever does call `html_safe` is going to need to be able to guarantee the string is safe, even if it has been prepared with some untrusted input.

Maybe the right place to do that is Elaine, but hopefully in a way that doesn't rely on every caller to remember they need to do this. But I'm fine with whatever you decide. Just trusting that you've thought about it. Thanks!

Reply:
I think this is a good reason to keep the interface separate (`_th` instead of `_t`), which I think we're doing already? That way it's a little more obvious potentially that `_th` is special and more dangerous. 

I don't think we can do this within the gem, but basically I think it'll be up to the caller to specify within the lambda that the `chunks` still get escaped?

```
'Simple string with <A>tag</A>',
{  A: lambda { |content| "<b>#{CGI::Escape(content)}</b>" },
```

or something like that? Because we know the tags we're writing are safe, but the content inside them is almost certainly not safe. But the gem can't enforce that escaping in the lambdas?